apps/evm_family/evm_sign_msg.c

```diff
@@ -0,0 +1,436 @@
+ * request.
+ * - Get user confirmation on the wallet and coin for which flow is requested.
+ * - Copy the init request data to sign_msg_ctx.
+ *
+ * @param query Reference to the decoded query struct from the host app
+ *
+ * @return a boolean value.
+ */
+static bool handle_initiate_query(evm_query_t *query);
+
+/**
+ * @brief This function is responsible for retrieving and assembling message
+ * data chunks for signing.
+ *
+ * @param query Reference to the decoded query struct from the host app
+ *
+ * @return a boolean value indicating if the message data has been reconstructed
+ * from chunks correctly or not.
+ */
+static bool get_msg_data(evm_query_t *query);
+
+/**
+ * @brief This function checks the message type and displays the message data
+ * for verification.
+ *
+ * @return a boolean indicating user verification or the rejection.
+ */
+static bool get_user_verification();
+
+/**
+ * @brief This function generates a signature for a message using a given
+ * derivation path and private key.
+ *
+ * @param sig The parameter `sig` is a pointer to a structure of type
+ * `evm_sign_msg_signature_response_t`.
+ *
+ * @return a boolean value indicating if signautre for msg data is generated
+ * correctly or not.
+ */
+static bool get_msg_data_signature(evm_sign_msg_signature_response_t *sig);
+
+/**
+ * @brief This function sends a signature response for a sign message query.
+ *
+ * @param query A pointer to an evm_query_t struct, which contains information
+ * about the query being processed.
+ * @param sig The parameter `sig` is of type
+ * `evm_sign_msg_signature_response_t`, which is a structure containing the
+ * signature response for a sign message request.
+ *
+ * @return a boolean value indicating if the signature is sent to the host
+ * correctly or not.
+ */
+static bool send_signature(evm_query_t *query,
+                           evm_sign_msg_signature_response_t *sig);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static evm_sign_msg_context_t sign_msg_ctx;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+static bool check_which_request(const evm_query_t *query,
+                                pb_size_t which_request) {
+  if (which_request != query->get_public_keys.which_request) {
+    evm_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
+                   ERROR_DATA_FLOW_INVALID_REQUEST);
+    return false;
+  }
+
+  return true;
+}
+
+static bool validate_initiate_query(evm_sign_msg_initiate_request_t *init_req) {
+  switch (init_req->message_type) {
```

Comment:
can we remove the switch case? seems overkill

Reply:
considering

Reply:
I have the opposite feeling, it's easier to read and I don't want to do a comparison based on the enum range.

Reply:
we've discussed this and can be resolved